parquet-column/src/main/java/org/apache/parquet/schema/Type.java

```diff
@@ -114,6 +116,11 @@ public boolean isMoreRestrictiveThan(Repetition other) {
   private final ID id;
 
   /**
+   * Pattern for matching column names that need to wrapped in backquotes.
+   */
+  private static final Pattern pattern = Pattern.compile("[^a-zA-Z\\d_-]");
```

Comment:
@rdblue : does that satisfy your requirement?
Ideally this should describe only what was supported for parquet before that change.
We don't want to quote strings that worked before without quoting.
Should this be anything that is in MessageTypeParser.Tokenizer's constructor initialization of the delimiter list? " ,;{}()\n\t="
What are the other character that wouldn't work? I think "." since it is used as a column path delimiter.